spec/v1/deps/git.go

```diff
@@ -80,18 +80,26 @@ func (gs *Git) UnmarshalJSON(data []byte) error {
 
 // Name returns the repository in a go-like format (example.com/user/repo/subdir)
 func (gs *Git) Name() string {
+	// in case of explicitly specified repositories, remove the `.git` suffix
+	if strings.HasSuffix(gs.Repo, ".git") {
+		gs.Repo = strings.TrimSuffix(gs.Repo, ".git")
```

Comment:
```suggestion
		gs.Repo = strings.TrimSuffix(gs.Repo, ".git")
```

Sorry for the late review. "Getters" like this one should not modify their pointer receiver however. You should be good just inlining that `strings.TrimSuffix` into the `fmt.Sprintf` call.

Also please check that for the other getters :)